server/player_data.go

```diff
@@ -35,3 +36,17 @@ func (p *PlayerData) addBlame(verificationKey string) bool {
 
 	return true
 }
+
+func (p *PlayerData) String() string {
+	return fmt.Sprintf(""+
+		"vk:%s, "+
+		"ip:%s"+
+		"pool:%d, "+
+		"num:%d, "+
+		"blames:%d "+
+		"amount:%d "+
+		"version:%d "+
+		"isPassive:%t\n",
+		p.verificationKey, getIP(p.conn), p.pool.num, p.number, len(p.blamedBy),
+		p.amount, p.version, p.isPassive)
+}
```

Comment:
This could make some log lines very long but I think it is worth trying.

If I should be parsing into JSON or something else, please let me know.

Reply:
Anything is better than nothing.  I wouldn't bother with JSON.  The logs are going to be designed to be scanned visually.  Long lines are preferable in my mind.

Reply:
Personally I think this is overkill. If a dev wants to print the PlayerData they can do that manually. The other logs I find useful, but this is very noisy.

Reply:
Also why only commas for some of them?

Reply:
I shortened this down to:

```
func (p *PlayerData) String() string {
	return fmt.Sprintf("("+
		"vk:%s, "+
		"pool:%d, "+
		"num:%d, "+
		"ip:%s, "+
		")",
		p.verificationKey, p.pool.num, p.number, getIP(p.conn))
}
```

Still too noisy? vk is nice as a unique value. IP is nice because it allows correlation with connection logs (where player data is not available). Pool and num could arguably be removed but they are helpful for correlating when a pool may have an issue.

Also, I confirmed that this String only appears in debug logs. It shouldn't be appearing anywhere else.

Reply:
> Also why only commas for some of them?

The missing commas, spaces and newlines were just bad formatting. Should be fixed now.

Reply:
This looks good, have we considered multi line log lines? Then we can do:

```
Something something something happend
Player: <output of PlayerData.String()
```

Reply:
That would probably help readability wherever players are logged.

Do these look like you are imagining? c90360d

```
DEBU[0000] [Blame] Duplicate blame
From player: (vk:7, pool:1, num:4, ip:)
To player: (vk:4, pool:1, num:1, ip:) 

DEBU[0000] [Blame] Blame applied for reason: LIAR
From player: (vk:8, pool:1, num:5, ip:)
To player: (vk:4, pool:1, num:1, ip:) 

DEBU[0000] [Ban] User blamed out of round
Player: (vk:4, pool:1, num:1, ip:) 

DEBU[0000] [Broadcast] 
From player: (vk:18, pool:1, num:3, ip:) 

DEBU[0000] [Ban] Disconnecting passive player
Player: (vk:16, pool:1, num:1, ip:)
```

Reply:
Yep that looks great. =)